colcon_core/task/__init__.py

```diff
@@ -299,3 +301,25 @@ def install(args, rel_src, rel_dst):
             shutil.rmtree(dst)
         if not os.path.exists(dst):
             os.symlink(src, dst)
+
+
+def collect_existing_environment_hooks(path):
+    """
+    Get a list of a all existing environment hooks in the given directory.
+
+    :param Path path: The path to check for existing environment hooks
+    :rtype: list
+    """
+    additional_hooks = []
+    shell_extensions = get_shell_extensions()
+    file_extensions = OrderedDict()
+    for shell_extensions_same_prio in shell_extensions.values():
+        for shell_extension in shell_extensions_same_prio.values():
+            for file_extension in shell_extension.get_file_extensions():
+                file_extensions[file_extension] = shell_extension
```

Comment:
`shell_extension` is not used  below. What about making `file_extensions` a set?

```suggestion
            file_extensions.update(shell_extension.get_file_extensions())
```